lib/components/mobile/batch-search-screen.tsx

```diff
@@ -44,32 +75,86 @@ class BatchSearchScreen extends Component<Props> {
-              onTextInputClick={this._fromFieldClicked}
-              selfValidate={planTripClicked}
-              showClearButton={false}
-            />
-            <LocationField
-              inputPlaceholder={intl.formatMessage({
-                id: 'components.LocationSearch.setDestination'
-              })}
-              isRequired
-              locationType="to"
-              onTextInputClick={this._toFieldClicked}
-              selfValidate={planTripClicked}
-              showClearButton={false}
-            />
-            <div className="switch-button-container-mobile">
-              <SwitchButton />
-            </div>
-            <BatchSettings onPlanTripClick={this.handlePlanTripClick} />
-          </div>
+          <MobileSearchSettings
+            advancedPanelOpen={showAdvancedModeSettings}
+            className={`batch-search-settings mobile-padding ${
+              showAdvancedModeSettings && 'advanced-mode-open'
+            }`}
+          >
+            <TransitionStyles transitionDelay={transitionDelay}>
+              <TransitionGroup style={{ display: 'content' }}>
+                {!showAdvancedModeSettings && (
+                  <CSSTransition
+                    classNames={mainPanelClassName}
+                    nodeRef={this._mainPanelContentRef}
+                    timeout={transitionDuration}
+                  >
+                    <div
+                      ref={this._mainPanelContentRef}
+                      style={{ display: 'content' }}
+                    >
+                      <LocationField
+                        inputPlaceholder={intl.formatMessage({
+                          id: 'components.LocationSearch.setOrigin'
+                        })}
+                        isRequired
+                        locationType="from"
+                        onTextInputClick={this._fromFieldClicked}
+                        selfValidate={planTripClicked}
+                        showClearButton={false}
+                      />
+                      <LocationField
+                        inputPlaceholder={intl.formatMessage({
+                          id: 'components.LocationSearch.setDestination'
+                        })}
+                        isRequired
+                        locationType="to"
+                        onTextInputClick={this._toFieldClicked}
+                        selfValidate={planTripClicked}
+                        showClearButton={false}
+                      />
+                      <div className="switch-button-container-mobile">
+                        <SwitchButton />
+                      </div>
+                      <BatchSettings
+                        onPlanTripClick={this.handlePlanTripClick}
+                        openAdvancedSettings={
+                          () =>
+                            this.setState({ showAdvancedModeSettings: true })
+                          // eslint-disable-next-line react/jsx-curly-newline
```

Comment:
Might need to `useCallback` for these, so you can also remove the eslint exception. (multiple instances)

Reply:
Since this (and other instances) is in a class component, `useCallback` isn't permitted. Is there an alternative for this?

Reply:
Oh, yes, if it's a component class, then just extract a class method (so that the function isn't recreated on each render).

Reply:
Gotcha, fixed in [0778c58](https://github.com/opentripplanner/otp-react-redux/pull/1249/commits/0778c58e3cf601a67e34fb54e3702a790b320146)